q2_demux/_summarize/_visualizer.py

```diff
@@ -0,0 +1,144 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import collections
+import os
+import pkg_resources
+import shutil
+import random
+
+import pandas as pd
+import seaborn as sns
+import numpy as np
+
+from q2_demux._demux import _read_fastq_seqs
+import q2templates
+
+TEMPLATES = pkg_resources.resource_filename('q2_demux', '_summarize')
+
+
+def _decode_qual_to_phred(qual_str):
+    # this function is adapted from scikit-bio
+    qual = np.fromstring(qual_str, dtype=np.uint8) - 33
+    return qual
+
+
+# TODO: Remove _PlotQualView once QIIME 2 #220 completed
+class _PlotQualView:
+    """
+    A very simple pass-through view which is made up of a single-end or
+    paired-end directory format with a bool indicating if single or paired.
+    """
+    def __init__(self, directory_format, paired):
+        self.directory_format = directory_format
+        self.paired = paired
+
+
+def summarize(output_dir: str, data: _PlotQualView, n: int=10) -> None:
+    paired = data.paired
+    data = data.directory_format
+
+    manifest = pd.read_csv(os.path.join(str(data), data.manifest.pathspec),
+                           header=0, comment='#')
+    manifest.filename = manifest.filename.apply(
+        lambda x: os.path.join(str(data), x))
+
+    fwd = manifest[manifest.direction == 'forward'].filename.tolist()
+    rev = manifest[manifest.direction == 'reverse'].filename.tolist()
+
+    per_sample_fastq_counts = {}
+    for file in fwd:
+        count = 0
+        for seq in _read_fastq_seqs(file):
+            count += 1
+        sample_name = os.path.basename(file).split('_', 1)[0]
+        per_sample_fastq_counts[sample_name] = count
+
+    result = pd.Series(per_sample_fastq_counts)
+    result.name = 'Sequence count'
+    result.index.name = 'Sample name'
+    result.sort_values(inplace=True, ascending=False)
+    result.to_csv(os.path.join(output_dir, 'per-sample-fastq-counts.csv'),
+                  header=True, index=True)
+
+    quality_scores = collections.defaultdict(list)
+    subsample_ns = sorted(random.sample(range(1, result.sum() + 1), n))
+    target = subsample_ns.pop(0)
```

Comment:
Oops, yeah I meant to go back at this and do a refactor, this was the "get it working" sampling method. Since we have the counts of each sample, it can actually skip entire files based on the counts if need be.